src/main/kotlin/sc/iview/ui/SwingMainWindow.kt

```diff
@@ -58,228 +59,239 @@ class SwingMainWindow(val sciview: SciView) : MainWindow {
+            } catch (e: HeadlessException) {
+                x = 10
+                y = 10
+            }
 
-        frame = JFrame("sciview")
-        frame.layout = BorderLayout(0, 0)
-        frame.setSize(sciview.windowWidth, sciview.windowHeight)
-        frame.setLocation(x, y)
-
-        splashLabel = SplashLabel()
-        val glassPane = frame.glassPane as JPanel
-        glassPane.isVisible = true
-        glassPane.layout = BorderLayout()
-        glassPane.isOpaque = false
-        glassPane.add(splashLabel, BorderLayout.CENTER)
-        glassPane.requestFocusInWindow()
-        glassPane.revalidate()
-
-        frame.defaultCloseOperation = JFrame.DISPOSE_ON_CLOSE
-        nodePropertyEditor = SwingNodePropertyEditor(sciview)
-
-        sceneryJPanel = SceneryJPanel()
-        JPopupMenu.setDefaultLightWeightPopupEnabled(false)
-        val swingMenuBar = JMenuBar()
-        val menus = sciview.scijavaContext?.getService(MenuService::class.java) ?: throw IllegalStateException("MenuService not available")
-        SwingJMenuBarCreator().createMenus(menus.getMenu("SciView"), swingMenuBar)
-
-        val bar = ProgressPie()
-        bar.value = 0.0
-        bar.minimumSize = Dimension(30, 30)
-        bar.maximumSize = Dimension(30, 30)
-        bar.preferredSize = Dimension(30, 30)
-        val progressLabel = JLabel("<html><strong></strong></html>")
-        progressLabel.horizontalAlignment = SwingConstants.RIGHT
-        swingMenuBar.add(Box.createHorizontalGlue())
-        swingMenuBar.add(progressLabel)
-        swingMenuBar.add(bar)
-        frame.jMenuBar = swingMenuBar
-
-        sciview.taskManager.update = { current ->
-            if(current != null) {
-                progressLabel.text = "<html><strong>${current.source}:</strong> ${current.status} </html>"
-                bar.value = current.completion.toDouble()
-            } else {
-                progressLabel.text = ""
-                bar.value = 0.0
+            frame = JFrame("sciview")
+            frame.layout = BorderLayout(0, 0)
+            frame.setSize(sciview.windowWidth, sciview.windowHeight)
+            frame.setLocation(x, y)
+
+            splashLabel = SplashLabel()
+            val glassPane = frame.glassPane as JPanel
+            glassPane.isVisible = true
+            glassPane.layout = BorderLayout()
+            glassPane.isOpaque = false
+            glassPane.add(splashLabel, BorderLayout.CENTER)
+            glassPane.requestFocusInWindow()
+            glassPane.revalidate()
+
+            frame.defaultCloseOperation = JFrame.DISPOSE_ON_CLOSE
+            nodePropertyEditor = SwingNodePropertyEditor(sciview)
+
+            sceneryJPanel = SceneryJPanel()
+            JPopupMenu.setDefaultLightWeightPopupEnabled(false)
+            val swingMenuBar = JMenuBar()
+            val menus = sciview.scijavaContext?.getService(MenuService::class.java)
+                    ?: throw IllegalStateException("MenuService not available")
```

Comment:
There is a shorter way to write this:
```kotlin
val menus = sciview.scijavaContext?.service(MenuService::class.java)
```
And then you don't need the conditional `IllegalStateException` because the `service` (rather than `getService`, which returns null) method does that for you already.

Reply:
Is it really possible for the `scijavaContext` to be null? If so, we should maybe change that...

Reply:
This is inherited from scenery, and there it might actually be null if it's not constructed (e.g. by the REPL)